src/linker/Linker/RemoveAttributeInstancesAttribute.cs

```diff
@@ -14,14 +16,14 @@ namespace Mono.Linker
 		AttributeTargets.Class, Inherited = false)]
 	public sealed class RemoveAttributeInstancesAttribute : Attribute
 	{
-		public RemoveAttributeInstancesAttribute ()
+		public RemoveAttributeInstancesAttribute (System.Collections.Generic.ICollection<CustomAttributeArgument> values)
 		{
-			Arguments = Array.Empty<CustomAttributeArgument> ();
-		}
-
-		public RemoveAttributeInstancesAttribute (CustomAttributeArgument value1)
-		{
-			Arguments = new[] { value1 };
+			// Arguments were previously boxed in Object, but don't need to be
+			Arguments = values.Select ((arg) => {
+				Debug.Assert (arg.Value is CustomAttributeArgument);
+				if (arg.Value is CustomAttributeArgument caa) return caa;
+				return arg;
```

Comment:
I don't understand this - the assert says that `arg.Value` would always be `CustomAttributeArgument` - but then we check for it again?
Is this to prevent runtime crashes if something goes wrong?

Reply:
Yeah, unless we do decide to support "unboxed" arguments to RemoveAttributeInstances, this doesn't make sense here.